react-components/hello-world.md

```diff
@@ -0,0 +1,386 @@
+# Hello, World!
+
+## Goals
+
+Our goal is to practice diving into a new library. By creating a Hello, World! project, we'll gain knowledge about how to start and run a React webapp.
+
+## Format
+
+To jump into front-end development with React, we will create a Hello, World app.
+
+We'll take the following steps:
+
+1. Introduce the tool that will create an initial React app on our machine
+1. Run a local server to look at our React app
+1. Read through React code, namely the `App` component
+
+## Using a Toolchain
+
+We will use a specific tool to help us begin a minimal React app: the [Create React App](https://create-react-app.dev/) tool.
+
+Create React App creates a new project, which includes starter HTML, CSS, and JS files, the React library, a recommended folder structure, and useful packages.
+
+To use Create React App, we use this command:
+
+```bash
+$ npx create-react-app hello-world
+```
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                            |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| `npx`                                               | A CLI tool that runs packages. In this curriculum, we will use this infrequently.                                                |
+| `create-react-app`                                  | A package that includes the command for Create React App, which creates a minimal React app.                                     |
+| `hello-world`                                       | **Replace this** with the name of your new React project. This will be the name of the project and the generated project folder. |
+
+If `npx` isn't available, we should follow directions to install the latest version of [Node.js](https://nodejs.org/en/). Node.js includes `npm`. `npm` 5.2+ includes `npx`.
+
+After running `create-react-app`, we'll have a new project folder. We should `cd` into our project folder before we do anything else!
+
+```bash
+$ cd hello-world
+```
+
+## Running A Server
+
+Let's look at what the `create-react-app` tool has given us!
+
+As part of using Create React App, we'll begin to use the `yarn` CLI tool. With `yarn` CLI, we'll be able to run scripts that are defined in `package.json`, a file also generated by Create React App. These scripts can run servers that load and reload our React app, which simplifies our development workflow.
+
+In our project root, we start a server with this command in the terminal:
+
+```bash
+$ yarn start
+```
+
+By default, our front-end is viewable at `localhost:3000`.
+
+Every time we make a change in a file and save it, our server will detect changes and then reload the app immediately.
+
```

Comment:
suggest warning about the dangers of autosave combined with tools that like to autoload.
```suggestion

### !callout-info

## VSCode Autosave, Meet React Autoload

It's often convenient to have our editor automatically save our files as we're working to help us avoid needless hours spent debugging an issue that was ultimately caused by forgetting to save a file. However, this help can become a hindrance when we are working with a program like React that watches for changes and then reloads. Since VSCode saves constantly, even while in the middle of typing a line of code, it's possible (actually common!) for React to try reloading our code when it's not syntactically correct, causing errors in our project.

<br />

When working with React, or other autoloading programs, it's a good idea to disable the VSCode autosave feature. Under the File menu, simply uncheck Auto Save!

### !end-callout

```